### ISSUE TITLE
Unhandled error

### ISSUE THREAD

**Issue by outlawtorn** [papandreou/impro#6]
Hi @papandreou @paras20xx 
We are getting this error in our logs which is not handled can you please help us on this one?

(commandArgs = err.commandArgs || this.usedEngines[i].commandArgs)
TypeError: Cannot read properties of undefined (reading 'commandArgs')

![image](https://github.com/papandreou/impro/assets/3257189/70f2332e-1469-4878-b03b-c53d5f4a4026)

**Comment by papandreou** [comment]
Hi! Looks like https://github.com/papandreou/express-processimage, and not this module? Could you open an issue there with reproduction steps?

**Comment by outlawtorn** [comment]
Hi @papandreou ok. Im still gathering data aand trying to reproduce it.

**Comment by alexjeffburke** [comment]
Hi there,

What you've reported here is kinda nasty.. it suggests that one streams registered as part of the Pipeline emitted something that was not an error instance (reminder that the fact this can happen is a real ugly anti-feature of JS semantics). This really should be debugged in whatever module whose stream it is allows that to happen.

That said, after a fair amount of thought/debate I decided to add some code to handle this situation. When I went through the exercise of heavily hardening the error handling the intent was to make sure this should be a reliably building block anywhere its used. Though your report comes from use within express-processimage I think this is worth hardening against for other callers given this is a situation we can check for.

A patch that should address this issue has been added in https://github.com/papandreou/impro/commit/fdb87a063798d06bb9a467d7ac7feb7985db8ef8. Note I'd still encourage pursuing more detail as suggested by @papandreou and attempting to fix the underlying cause of the non-Error emit.

Thanks, Alex J Burke.
